### PR TITLE
Preview your username colour as you type

### PR DESCRIPTION
**Pull request by urkerab**
This is loosely based on the first suggestion from http://www.smogon.com/forums/posts/6424903 which asked for a link to the colour preview. I thought it would be neater simply to preview the username as you choose it.

### CODE REVIEW

**Comment by Zarel** [comment]
If we do this, we might as well bold the username?

**Comment by Zarel** [comment]
I don't think we colorize usernames anywhere where we don't also bold the username, for readability reasons.

**Comment by ascriptmaster** [comment]
Where does the color change? In the input field as you type? If that's the case I'd rather have a box that changes color as you type. Otherwise, it'll depend where it's changing.

Also, I think users without room staff aren't bold in the user list.

**Comment by urkerab** [comment]
Well, I thought the input field was the most obvious place to change the colour as it's already got the username in it.

**Comment by Zarel** [comment]
Well, I still have worries that it would be distracting. Maybe the border color of the box would be less distracting?

**Comment by ascriptmaster** [comment]
I'd absolutely loathe it if the color of the text changed as I was typing it. Border or box is preferred, or if you *really wanted* I guess you could have an extra line that displayed the color w/ the name like the color preview does, but I'd prefer you didn't because that's a lot of extra space.

**Comment by Zarel** [comment]
You could just have a bullet somewhere, and colorize that, like the bullet in `/me`.

**Comment by urkerab** [comment]
Colourised bullet it is.

**Comment by Zarel** [comment]
Screenshot?

**Comment by urkerab** [comment]
username

**Comment by urkerab** [comment]
Oops, bug crept into 59c06fd.

**Comment by Zarel** [comment]
Maybe right-align the bullet.

Maybe even have it right-aligned inside the textbox.

...my next fear is that people will see a red bullet and think it means "invalid" instead of "your username color will be red".

Maybe instead of a bullet, just the text "COLOR" in bold small font.

**Comment by urkerab** [comment]
How about this version?
username

**Comment by Zarel** [comment]
Looks really weird, especially having the name twice. I think just having it right-aligned and saying "COLOR" or "(color)" might be better.

**Comment by urkerab** [comment]
username

File: style/client.css [hunk 1]
```diff
@@ -2668,6 +2668,10 @@ a.ilink:hover {
 	color: #555555;
 }
 
+.bull {
+	float: right;
```
Comment (Zarel) [inline style/client.css]: This need a decreased font size, as well. Looks a bit ugly at its current size.

**Comment by urkerab** [comment]
username
Also renamed the CSS class as I'm no longer using a `&bull;`.

**Comment by DanUgore** [comment]
Damn I'm late but I think having the name twice was a good idea :(

**Pull request merged**